### ISSUE TITLE
Errors in a @constraints block reported with incorrect line numbers

### ISSUE THREAD

**Issue by niclasmattsson** [jump-dev/JuMP.jl#1174]
If you make an error in a `@constraints` block the line number reported to cause the error is the line of the `@constraints` statement. For example:
```julia
m = Model(solver=ClpSolver())

@variables m begin
    0 <= x <= 4
    0 <= y <= 30
end

@constraints m begin        # <- error reported on this line
    region1, 1x + 5y <= 3.0
    region2, x - undeclared_variable <= 2.0
end

@objective(m, Max, 5x + 3y)

print(m)
status = solve(m)
```

In this example the error is `ERROR: LoadError: UndefVarError: undeclared_variable not defined`, so you would easily find the problem anyway. But I often get more generic errors which don't have an immediately obvious cause, and if you have 20+ constraints then it becomes extremely difficult to identify the problem.

This greatly limits the usability of the `@constraints` syntax for large models - the precise situation in which it is intended to be convenient.

**Comment by mlubin** [comment]
Thanks for the report. This should be fixed for JuMP 1.0 but has less priority than the ongoing MOI transition.

**Comment by niclasmattsson** [comment]
Excellent! No hurry, I want the MOI stuff and the changes to JuMPcontainers done before this too. :) But that said I'm glad to see you tagged this for the 1.0 release.
